### PR TITLE
Added the ability to send action bars

### PR DESCRIPTION
**Pull request by KCodeYT**

### CODE REVIEW

**Comment by XDqrkneZ** [comment]
merge?

File: gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java [hunk 1]
```diff
@@ -1276,6 +1276,22 @@ public void sendTitle( String title, String subtitle ) {
...
+        titlePacket.setText( message );
+        titlePacket.setFadeInTime( (int) unit.toMillis( fadein ) / 50 );
+        titlePacket.setStayTime( (int) unit.toMillis( duration ) / 50 );
+        titlePacket.setFadeOutTime( (int) unit.toMillis( fadeout ) / 50 );
+        this.getConnection().addToSendQueue( titlePacket );
+    }
+
+    @Override
+    public void sendActionbar( String message ) {
+        this.sendActionbar( message, 1, 1, (long) 0.5, TimeUnit.SECONDS );
```
Comment (Janmm14) [inline gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java:1292]: wtf why is here 0.5 casted to a long that looks senseless (also for the title)

File: gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java [hunk 2]
```diff
@@ -296,6 +307,8 @@
      */
     void sendTitle( String title, String subtitle );
 
+    void sendActionbar(String message);
```
Comment (geNAZt) [inline gomint-api/src/main/java/io/gomint/entity/EntityPlayer.java:310]: Please add javadoc

File: gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java [hunk 3]
```diff
@@ -1276,6 +1276,22 @@ public void sendTitle( String title, String subtitle ) {
         this.sendTitle( title, subtitle, 1, 1, (long) 0.5, TimeUnit.SECONDS );
     }
 
+    @Override
+    public void sendActionbar( String message, long fadein, long duration, long fadeout, TimeUnit unit ) {
+        PacketSetTitle titlePacket = new PacketSetTitle();
+        titlePacket.setType( PacketSetTitle.TitleType.TYPE_ACTION_BAR.getId() );
+        titlePacket.setText( message );
+        titlePacket.setFadeInTime( (int) unit.toMillis( fadein ) / 50 );
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java:1284]: Cast the result to an int not the first operator

File: gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java [hunk 4]
```diff
@@ -1276,6 +1276,22 @@ public void sendTitle( String title, String subtitle ) {
...
+        titlePacket.setStayTime( (int) unit.toMillis( duration ) / 50 );
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java:1285]: Cast the result to an int not the first operator

File: gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java [hunk 5]
```diff
@@ -1276,6 +1276,22 @@ public void sendTitle( String title, String subtitle ) {
...
+        titlePacket.setFadeOutTime( (int) unit.toMillis( fadeout ) / 50 );
```
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java:1286]: Cast the result to an int not the first operator

File: gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java (hunk 1 above)
Comment (geNAZt) [inline gomint-server/src/main/java/io/gomint/server/entity/EntityPlayer.java:1292]: Please use a more proper unit (Milliseconds) to avoid casting

**Review by geNAZt — CHANGES_REQUESTED** [review]

**Comment by lukeeey** [comment]
I did this in my commands PR too and also changed the method for sending the animation times.

~~Also i think actionbar sending was already in gomint? I thought i just changed it in my PR~~

**Comment by geNAZt** [comment]
Oh then that has been slipped through, will look into that now

**Comment by lukeeey** [comment]
@geNAZt Nevermind, i was wrong.

**Pull request closed without merging**